core/src/main/java/com/datastrato/gravitino/storage/relational/service/SchemaMetaService.java

```diff
@@ -0,0 +1,255 @@
+        Objects.equals(oldSchemaEntity.id(), newEntity.id()),
+        "The updated schema entity id: %s should be same with the schema entity id before: %s",
+        newEntity.id(),
+        oldSchemaEntity.id());
+    Integer updateResult;
+    try {
+      updateResult =
+          SessionUtils.doWithCommitAndFetchResult(
+              SchemaMetaMapper.class,
+              mapper ->
+                  mapper.updateSchemaMeta(
+                      POConverters.updateSchemaPOWithVersion(
+                          oldSchemaPO, newEntity, metalakeId, catalogId),
+                      oldSchemaPO));
+    } catch (RuntimeException re) {
+      if (re.getCause() != null
+          && re.getCause().getCause() != null
+          && re.getCause().getCause() instanceof SQLIntegrityConstraintViolationException) {
+        // TODO We should make more fine-grained exception judgments
+        // Usually throwing `SQLIntegrityConstraintViolationException` means that
+        // SQL violates the constraints of `primary key` and `unique key`.
+        // We simply think that the entity already exists at this time.
+        throw new EntityAlreadyExistsException(
+            String.format("Schema entity: %s already exists", newEntity.nameIdentifier()));
+      }
+      throw re;
+    }
+
+    if (updateResult > 0) {
+      return newEntity;
+    } else {
+      throw new IOException("Failed to update the entity: " + identifier);
+    }
+  }
+
+  public boolean deleteSchema(NameIdentifier identifier, boolean cascade) {
+    NameIdentifier.checkSchema(identifier);
+    String metalakeName = identifier.namespace().level(0);
+    String catalogName = identifier.namespace().level(1);
+    String schemaName = identifier.name();
+    Long metalakeId =
+        SessionUtils.getWithoutCommit(
+            MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeIdMetaByName(metalakeName));
+    if (metalakeId == null) {
+      throw new NoSuchEntityException(NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE, metalakeName);
+    }
+
+    Long catalogId =
+        SessionUtils.getWithoutCommit(
+            CatalogMetaMapper.class,
+            mapper -> mapper.selectCatalogIdByMetalakeIdAndName(metalakeId, catalogName));
+    if (catalogId == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE, identifier.namespace().toString());
+    }
+
+    Long schemaId =
+        SessionUtils.getWithoutCommit(
+            SchemaMetaMapper.class,
+            mapper -> mapper.selectSchemaIdByCatalogIdAndName(catalogId, schemaName));
+    if (schemaId != null) {
+      if (cascade) {
+        SessionUtils.doMultipleWithCommit(
+            () ->
+                SessionUtils.doWithoutCommit(
+                    SchemaMetaMapper.class,
+                    mapper -> mapper.softDeleteSchemaMetasBySchemaId(schemaId)),
+            () -> {
+              // TODO We will cascade delete the metadata of sub-resources under the catalog
```

Comment:
`under the catalog` -> `under the schema`

Reply:
Done